src/re/multi_grid/indexing.py

```diff
@@ -0,0 +1,576 @@
+#!/usr/bin/env python3
+
+import operator
+from dataclasses import dataclass
+from functools import reduce
+from typing import Callable, Optional, Iterable
+
+import numpy as np
+import numpy.typing as npt
+
+
+@dataclass()
```

Comment:
if you are using data classes you should not then define an __init__

instead of the __initi__ uses the __post_init__ magic method :-)

Reply:
Thanks for your comment.

As far as I know there is no recommendation for using one over the other. Both are the same length in this case and IMHO just defining init is more readable.